src/common/libsubprocess/remote.c

```diff
@@ -543,11 +545,63 @@ int subprocess_remote_setup (flux_subprocess_t *p, const char *service_name)
     return 0;
 }
 
-static int remote_output (flux_subprocess_t *p,
-                          const char *stream,
-                          const char *data,
-                          int len,
-                          bool eof)
+static int remote_output_local_unbuf (flux_subprocess_t *p,
+                                      const char *stream,
+                                      const char *data,
+                                      int len,
+                                      bool eof)
+{
+    struct subprocess_channel *c;
+
+    if (!(c = zhash_lookup (p->channels, stream))) {
+        llog_debug (p,
+                    "Error returning %d bytes received from remote"
+                    " subprocess pid %d %s: unknown channel name",
+                    len,
+                    (int)flux_subprocess_pid (p),
+                    stream);
+        errno = EPROTO;
+        set_failed (p, "error returning unknown channel %s", stream);
+        return -1;
+    }
+
+    if (data && len) {
+        c->unbuf_data = data;
+        c->unbuf_len = len;
+        c->unbuf_consumed = false;
+        if (eof)
+            c->read_eof_received = true;
+        c->unbuf_eof = false;
+
+        c->output_cb (c->p, c->name);
+
+        c->unbuf_consumed = false;
+    }
+    /* N.B. any data not consumed by the user is lost, so if eof is
+     * seen, we can send it in the local unbuf immediately */
+    if (eof) {
+        /* N.B. we technically should clear/memset c->unbuf_data
+         * before returning EOF, as a user could read from it during
+         * the eof callback.  But for performance reasons, we won't do
+         * that and assume the caller is smart enough not to do that.
+         */
+        c->read_eof_received = true;
+        c->unbuf_len = 0;
+        c->unbuf_eof = true;
```

Comment:
Why do we have both `c->eof_received` and `c->unbuf_eof`?

Reply:
I think `unbuf_eof` is left over from a prior attempt.  Should remove it.